CONTRIBUTING.md

```diff
@@ -0,0 +1,55 @@
+# Contributing
+## In short
+1. Clone the repository
+   ```
+   git clone https://github.com/Earthcomputer/clientcommands
+   cd clientcommands
+   ```
+1. Generate the Minecraft source code
+   ```
+   ./gradlew genSources
+   ```
+    - Note: on Windows, use `gradlew` rather than `./gradlew`.
+1. Import the project into your preferred IDE.
+    1. If you use IntelliJ (the preferred option), you can simply import the project as a Gradle project.
+    1. If you use Eclipse, you need to `./gradlew eclipse` before importing the project as an Eclipse project.
+1. Edit the code
+1. After testing in the IDE, build a JAR to test whether it works outside the IDE too
+   ```
+   ./gradlew build
+   ```
+   The mod JAR may be found in the `build/libs` directory
+1. [Create a pull request](https://help.github.com/en/articles/creating-a-pull-request)
+   so that your changes can be integrated into clientcommands
+    - Note: for large contributions, create an issue before doing all that
+      work, to ask whether your pull request is likely to be accepted
+## Scope
+clientcommands is not a hacked or cheat client. Commands or utilities whose sole purpose is to provide an unfair advantage over other players are out of scope for clientcommands.
+
+That said, commands which are only needed in very specific circumstances are also out of scope. clientcommands consists of commands that are generally useful. For edge cases there is the [scripting API](https://github.com/Earthcomputer/clientcommands/blob/fabric/docs/clientcommands.ts) contained within this mod.
+## Code style
+Please adhere to these styling instructions when contributing. If you don't do this initially you will be requested to change your commits accordingly.
+These instructions are not foundational. You're expected to have an understanding of what is clean Java code and what isn't. These instructions build on general consensus.
+### Variables
+#### Naming
+* Use descriptive variable names everywhere, except when they are meaningless (loop-indices etc.)
+#### Command exceptions
+* Exception variables should use SCREAMING_SNAKE_CASE
+* Mark exception variables as `private static final`
+* They should always be translated
```

Comment:
This can be generalized too. Any message that the user may end up seeing needs to be translated. The only exception to this is if the text is used for debugging purposes, e.g. logging or info on the f3 screen etc.